core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -336,6 +340,70 @@ public LogicalPlan visitAggregation(Aggregation node, AnalysisContext context) {
     return new LogicalAggregation(child, aggregators, groupBys);
   }
 
+  @Override
+  public LogicalPlan visitFieldSummary(FieldSummary node, AnalysisContext context) {
+    LogicalPlan child = node.getChild().getFirst().accept(this, context);
+
+    TypeEnvironment env = context.peek();
+    Map<String, ExprType> fieldsMap = env.lookupAllFields(Namespace.FIELD_NAME);
+
+    if (node.getIncludeFields() != null) {
+      List<String> includeFields =
+          node.getIncludeFields().stream()
+              .map(expr -> ((Field) expr).getField().toString())
+              .toList();
+
+      Map<String, ExprType> filteredFields = new HashMap<>();
+      for (String field : includeFields) {
+        if (fieldsMap.containsKey(field)) {
+          filteredFields.put(field, fieldsMap.get(field));
+        }
+      }
+      fieldsMap = filteredFields;
+    }
+
+    ImmutableList.Builder<NamedAggregator> aggregatorBuilder = new ImmutableList.Builder<>();
+    Map<String, Map.Entry<String, ExprType>> aggregatorToFieldNameMap = new HashMap<>();
+
+    for (Map.Entry<String, ExprType> entry : fieldsMap.entrySet()) {
+      ExprType fieldType = entry.getValue();
+      String fieldName = entry.getKey();
+      ReferenceExpression fieldExpression = DSL.ref(fieldName, fieldType);
+
+      aggregatorBuilder.add(new NamedAggregator("Count" + fieldName, DSL.count(fieldExpression)));
+      aggregatorToFieldNameMap.put("Count" + fieldName, entry);
+      aggregatorBuilder.add(
+          new NamedAggregator("Distinct" + fieldName, DSL.distinctCount(fieldExpression)));
+      aggregatorToFieldNameMap.put("Distinct" + fieldName, entry);
+
+      if (ExprCoreType.numberTypes().contains(fieldType)) {
+        aggregatorBuilder.add(new NamedAggregator("Max" + fieldName, DSL.max(fieldExpression)));
+        aggregatorToFieldNameMap.put("Max" + fieldName, entry);
+        aggregatorBuilder.add(new NamedAggregator("Min" + fieldName, DSL.min(fieldExpression)));
+        aggregatorToFieldNameMap.put("Min" + fieldName, entry);
+        aggregatorBuilder.add(new NamedAggregator("Avg" + fieldName, DSL.avg(fieldExpression)));
+        aggregatorToFieldNameMap.put("Avg" + fieldName, entry);
+      }
+    }
+
+    ImmutableList<NamedAggregator> aggregators = aggregatorBuilder.build();
+    ImmutableList<NamedExpression> groupBys = new ImmutableList.Builder<NamedExpression>().build();
+
+    // new context
+    context.push();
+    TypeEnvironment newEnv = context.peek();
+
+    newEnv.define(new Symbol(Namespace.FIELD_NAME, "Field"), ExprCoreType.STRING);
```

Comment:
consider encapsulating all the types and lists for FieldSummary within a class, like a `FieldSummaryAggregationExpression`.  Then you can define enums for each column, and builders for the analyzer and parser.